src/Core/Configure/Engine/JsonConfig.php

```diff
@@ -76,7 +76,7 @@ public function read(string $key): array
     {
         $file = $this->_getFilePath($key, true);
 
-        $values = json_decode(file_get_contents($file), true);
+        $values = json_decode((string)file_get_contents($file), true);
```

Comment:
Failure to load file contents should throw an exception as you have done in `IniConfig`. Currently one would get a unclear message stating JSON parsing failed when in fact the file itself could not be read.

Reply:
Ok in 5.x we can add those exception now